consensus/src/round_manager.rs

```diff
@@ -354,14 +355,31 @@ impl RoundManager {
         &mut self,
         new_round_event: NewRoundEvent,
     ) -> anyhow::Result<()> {
+        let is_valid_proposer = self
+            .proposer_election
+            .is_valid_proposer(self.proposal_generator.author(), new_round_event.round);
+
         counters::CURRENT_ROUND.set(new_round_event.round as i64);
         counters::ROUND_TIMEOUT_MS.set(new_round_event.timeout.as_millis() as i64);
         match new_round_event.reason {
             NewRoundReason::QCReady => {
                 counters::QC_ROUNDS_COUNT.inc();
             },
-            NewRoundReason::Timeout(_) => {
+            NewRoundReason::Timeout(ref reason) => {
                 counters::TIMEOUT_ROUNDS_COUNT.inc();
+                counters::ROUND_TIMEOUT_REASON
+                    .with_label_values(&[&reason.to_string(), &is_valid_proposer.to_string()])
+                    .inc();
+                if let RoundTimeoutReason::PayloadUnavailable { missing_authors } = reason {
+                    let ordered_peers = self.epoch_state.verifier.get_ordered_account_addresses();
+                    for idx in missing_authors.iter_ones() {
+                        if let Some(author) = ordered_peers.get(idx) {
+                            counters::ROUND_TIMEOUT_REASON_MISSING_AUTHORS
+                                .with_label_values(&[author.short_str().as_str()])
```

Comment:
this is aggregated missing authors? do we want to monitor the raw metrics of individual missing authors? we should be able to aggregate on grafana easily?

Reply:
we can collect raw as well, but we will need to monitor this aggregation anyhow to be precise. This aggregation looks at a specific quorum, not possible with grafana. I also wanted to make sure it's recorded by the leader so we don't explode cardinality.